test/libsolidity/semanticTests/viaYul/dirty_memory_dynamic_array.sol

```diff
@@ -0,0 +1,19 @@
+contract C {
+    function f() public pure returns (bool correct) {
+        uint8[] memory m = new uint8[](1);
+        m[0] = uint8(0x0101);
+        assembly {
+            mstore(m, 257)
+        }
+        uint8 x = m[0];
+        bool a = (m[0] == 0x01);
```

Comment:
The comparison might perform another cleanup. Can you change the test so that you return the value as a `uint` and use inline assembly to copy from `x` to the return variable?